### DISCUSSION TITLE
Flask-SocketIo | After logging in Chrome dosen't return session['name'] value but Firefox does.

### DISCUSSION THREAD

**Discussion by KarteekAti** [miguelgrinberg/Flask-SocketIO#1716] [Q&A]
When I log in using User-id Chrome doesn't return the Users name using session['name'], which I provided using the flasks session. But firefox does.

[**Chrome Screenshot**][1]

 

[**Firefox Screenshot**][2]

**Python**   

     @app.route('/get_name',methods=['GET'])
    def get_name():
    	data = {'name' : ''}
    	if NAME_KEY in session:
    		data = {'name':session['name']}
    	return jsonify(data)	
**JS**

    async function get_Name(){
        return fetch("/get_name")
        .then(function (response) {
            return  response.json();   
           })
           .then(function (text) {  
             return text["name"];
           });
    }
    socket.on('connect', async function() {
    var usr_name =  await get_Name();
    console.log(usr_name);
    if(usr_name != ''){
        var data = {
            'name': usr_name,
            'msg': 'User has connected!'
        };
        socket.emit('message',data);	
    }

Chrome returns an empty username, whereas Firefox returns logged username

  [1]: https://i.stack.imgur.com/cQWAJ.png
  [2]: https://i.stack.imgur.com/SmWzT.png

**Comment by miguelgrinberg** [comment]
I'm not sure if this is the problem or not, but your `get_name()` function looks very strange to me. Here is a correct version of it:

```javascript
async function get_Name(){
    const response = await fetch("/get_name");
    const text = await response.json();
    return text["name"];
}
```

You may want to add a print statement in the server to verify if the server has the correct name in the session or not. If it does, then this is a problem with your front end, unrelated to Flask-SocketIO.

**Reply by KarteekAti** [reply]
replaced get_Name with your code, still facing the same issue.

**Reply by miguelgrinberg** [reply]
I'm sorry but you are providing no additional information, so there is nothing else I can tell you. If you think this is a bug in this package, then build a complete application with server and client and I'll test it here.

**Reply by KarteekAti** [reply]
https://chatroom-private.herokuapp.com/home here's the link.

**Reply by KarteekAti** [reply]
Sorry, but  I have started making some changes to server. Trying using Ajax.

**Reply by miguelgrinberg** [reply]
You don't understand. If (only if) you think this is a bug in this package, then please create a short example that reproduces the problem and send me the source code so that I can investigate and fix the bug.

**Reply by KarteekAti** [reply]
Oh, ok my bad. I'm not sure is this related to front-end or Socketio. I'll try to investigate more.
